### PR TITLE
Create installation script for grafana

### PR DESCRIPTION
**Pull request by ChadNedzlek**
This installation script will
  - setup the grafana package repository
  - install/update grafana and pip
  - pip install dependencies for vault-env.py
  - set the grafana configuration to source code
  - override grafan-server settings to pull secrets from keyvault and put in environment

### CODE REVIEW

**Comment by ChadNedzlek** [comment]
@sunandabalu This is my new "service" deployment for setting up grafana. I expect it to run very, very rarely (we will probably never have to run this again after the next week or two). Take a peek for me and let me know if there is anything else deploymenty you'd like to check

@garath This should match the configuration we've got right now for grafana, and I believe it is idempotent... it can be run multiple times and will update the configuration to look like we want from source.

I haven't done any sort of automated restarting of the service to fetch new secrets... figured that would be a weekly cron task to just call "systemctl restart grafana-server", which will pick up the new secrets.

**Comment by garath** [comment]
Is grafana.ini fully reduced from the default?

File: src/Monitoring/grafana-init/grafana.ini [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# This setting is ignored if multiple OAuth providers are configured.
+;oauth_auto_login = false
+
+#################################### Generic OAuth ##########################
+[auth.generic_oauth]
+enabled = true
+name = Azure AD
+allow_sign_up = true
+client_id = a2541735-8225-40af-9c8a-2ae233203739
+;client_secret = <<managed by environment>
```
Comment (garath) [inline src/Monitoring/grafana-init/grafana.ini]:
```suggestion
;client_secret = <managed by environment>
```

**Comment by ChadNedzlek** [comment]
@garath It's only the settings we want to override from the default.  Figured that made more sense than duplicating all the defaults here (either commented out or re-assigning the existing default)

File: src/Monitoring/grafana-init/grafana.ini [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#################################### Generic OAuth ##########################
+[auth.generic_oauth]
+enabled = true
+name = Azure AD
+allow_sign_up = true
+client_id = a2541735-8225-40af-9c8a-2ae233203739
+;client_secret = <managed by environment>
+scopes = openid email name
+auth_url = https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db$
+token_url = https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011d$
```
Comment (garath) [inline src/Monitoring/grafana-init/grafana.ini]: Are these URLs right? They seem truncated.

Comment (ChadNedzlek) [reply src/Monitoring/grafana-init/grafana.ini]: They are indeed. Copy/pasting through putty is a mess. But I think I fixed them.

**Comment by garath** [comment]
Thoughts on setting GF_SECURITY_ADMIN_USER and/or GF_SECURITY_ADMIN_PASSWORD? Local logins are disabled, so I'm not sure it's even possible to log in under this account. Also, it's only used at first run to initialize the user. But, it is an automatic special admin user in Grafana.

**Comment by ChadNedzlek** [comment]
Yeah, it's just the default password.  As soon as it's used, it's forced to be changed, so it would only matter if someone beat us to the punch and logged in before we could.  But there wouldn't be much they could really do. (And we'd notice really fast, because "admin/admin" wouldn't work).

I'm not even sure it would be usable as is, because basic auth and the form are disabled here.  (We'd have to temporarily reenable the form when setting up a new machine to log in and make the first AS account admin).  It seems like more work to manage a "password" in keyvault that wouldn't even work except for initial login than to deal with the situation.

**Comment by garath** [comment]
Upgrading the Grafana server version should probably also upgrade the installed plugins. [Reference](https://grafana.com/docs/installation/upgrading/#update-plugins).

```
grafana-cli plugins update-all
```

**Comment by garath** [comment]
And also backup the database?

**Comment by ChadNedzlek** [comment]
The plugin update I can do.

I'm not sure how to backup the database, especially since a local backup isn't super useful... we'd need to get it off the machine somehow (since the only reason we'd need to restore would be because we lost the machine).  There isn't much in there that won't be deployed from source.  All that would be is the user mappings, which only really takes a few minutes to set up.

**Pull request merged**